src/registry/ecr.go

```diff
@@ -136,6 +139,18 @@ func (r *RegistryScan) GetLabelDigest(ctx context.Context, imageInfo ImageRefere
 	return digestInfo, nil
 }
 
+type WaiterError string
+
+func (w WaiterError) Error() string {
+	return string(w)
+}
+
+func IsErrWaiterTimeout(w error) bool {
+	return errors.As(w, &ErrWaiterTimeout)
```

Comment:
If we're just using the `bool` returned from As, would it be more appropriate to use `errors.Is` instead?

https://pkg.go.dev/errors#Is

Reply:
Oh yes